system/jlib/jfile.cpp

```diff
@@ -1868,6 +1868,27 @@ IFileIO * CFile::openShared(IFOmode mode,IFSHmode share,IFEflags extraFlags)
 
 //---------------------------------------------------------------------------
 
+static int globalFileSyncMaxRetrySecs = fileSyncRetryDisabled;
+static std::atomic<bool> globalFileSyncMaxRetrySecsConfigured{false};
+static CriticalSection globalFileSyncCS;
+static int getGlobalMaxFileSyncSecs()
+{
+    if (!globalFileSyncMaxRetrySecsConfigured)
+    {
+        CriticalBlock b(globalFileSyncCS);
+        if (!globalFileSyncMaxRetrySecsConfigured)
+        {
+            globalFileSyncMaxRetrySecsConfigured = true;
```

Comment:
This needs to be set at the end of the block - otherwise another thread could come in and read an uninitialised value...

Reply:
true, I'll change, although uninitializing it (due to an update) would still return the configured values, and only mean that it re-initialized on next request I think